xapian-core/api/valuerangeproc.cc

```diff
@@ -439,4 +439,82 @@ NumberRangeProcessor::operator()(const string& b, const string& e)
     return Xapian::Query(Xapian::Query::OP_INVALID);
 }
 
+static const string byte_units[4] = {
+    "B", "K", "M", "G"
+};
```

Comment:
It's better to avoid `std::string` for constant string data like this - there's a fair bit of overhead to creating a string object which you don't have with a string constant, and in a case like this it's likely the compiler will generate code that actually initialises the strings in the array at runtime, whereas really we want such data to be read-only data created entirely at compile time.

The obvious alternative of an array of `const char*` is also problematic in a shared library because each pointer in such an array requires a load-time relocation, and the more of those there are, the slower the library loads.  Because of that we try to avoid adding more load-time relocations, and to eliminate existing ones.

In this case you should be able to declare this as an array of char arrays:

```
static const char byte_units[4][2] = {
    "B", "K", "M", "G"
};
```

That's also a nicely compact representation - at least on x86-64 Linux it's just 8 bytes (8 because each entry has a trailing nul byte).

Reply:
Changes done accordingly.